asv_bench/benchmarks/timeseries.py

```diff
@@ -335,6 +335,48 @@ def time_timeseries_asof_single(self):
         self.ts.asof(self.dates[0])
 
 
+class timeseries_dataframe_asof(object):
+    goal_time = 0.2
+
+    def setup(self):
+        self.N = 10000
```

Comment:
these are too small make 100000

put nans at beginning and another one at the end